distributed/deploy/adaptive.py

```diff
@@ -28,49 +33,154 @@ class Adaptive(object):
     ...         """ Bring worker count up to n """
     ...     def scale_down(self, workers):
     ...        """ Remove worker addresses from cluster """
+
+    Notes
+    -----
+    Subclasses can override :meth:`Adaptive.should_scale_up` and
+    :meth:`Adaptive.should_scale_down` to control when the cluster should be
+    resized. The default implementation checks if there are too many tasks
+    per worker or too little memory available (see :meth:`Adaptive.needs_cpu`
+    and :meth:`Adaptive.needs_memory`).
+
+    :meth:`Adaptive.get_scale_up_kwargs` method controls the arguments passed to
+    the cluster's ``scale_up`` method.
     '''
-    def __init__(self, scheduler, cluster, interval=1000, startup_cost=1):
+
+    def __init__(self, scheduler, cluster, interval=1000, startup_cost=1,
+                 scale_factor=2):
         self.scheduler = scheduler
         self.cluster = cluster
         self.startup_cost = startup_cost
+        self.scale_factor = scale_factor
         self._adapt_callback = PeriodicCallback(self._adapt, interval,
                                                 self.scheduler.loop)
         self.scheduler.loop.add_callback(self._adapt_callback.start)
         self._adapting = False
 
+    def needs_cpu(self):
```

Comment:
Thoughts on other needs?  For example `needs_resources`?  Should we think about generalizing this or is this premature?